### PR TITLE
Check-Config command with good error messages

### PR DESCRIPTION
**Pull request by Aylr**
## What's Here

- a new command: `great_expectations check-config`
- It does some basic sanity checking on your config file by building a `DataContext` which now has
- Much better error messages that are passed on to the CLI. These are all tested hence the pile of yml fixtures.
- now archives and offers to make new config file

### Bonus

- A much better `relative_file_path` function yanked from dagster.

### CODE REVIEW

File: great_expectations/cli/cli.py [hunk 1]
```diff
@@ -321,6 +321,74 @@ def render(render_object):
...
+    try:
+        context = DataContext(
+            context_root_dir="{}/great_expectations/".format(target_directory))
+        config = context.get_project_config()
+        check_for_obsolete_config_file(config)
+        cli_message("Your config file appears valid!")
+    except (
+            ge_exceptions.InvalidConfigurationYamlError,
+            ge_exceptions.InvalidTopLevelConfigKeyError,
+            ge_exceptions.MissingTopLevelConfigKeyError,
```
Comment (eugmandel) [inline great_expectations/cli/cli.py:390]: In the typical case of check-config being executed in a pre-0.8.0 project, this is the message that is displayed:
```
You are missing a required key: `ge_config_version`
    Please ensure that it is in your config file.
```

While it is technically correct, I think we should treat ge_config_version as a special key  and interpret its absence as ge_config_version=0.

**Review by eugmandel — CHANGES_REQUESTED** [review]
Looks great - it is awesome to see the specific error types!

1. Can we include the offer to regenerate a new config file in this PR before we merge? Without it the user story does not really work, since the user has no path forward if the check does not pass.
2. See the inline comment on cli.py

File: great_expectations/cli/cli.py (hunk 1 above)
Comment (Aylr) [reply great_expectations/cli/cli.py:390]: 1. I think you are right to press me on this. I'll do it.
2. You are spot on with this one! I'll implement

Comment (eugmandel) [reply great_expectations/cli/cli.py:390]: @Aylr Thank you!

Comment (Aylr) [reply great_expectations/cli/cli.py:390]: Both are implemented and tested.

**Review by eugmandel — APPROVED** [review]

**Pull request merged**